consensus/safety-rules/src/tests/vault.rs

```diff
@@ -6,22 +6,38 @@ use libra_crypto::{ed25519::Ed25519PrivateKey, Uniform};
 use libra_secure_storage::{KVStorage, Storage, VaultStorage};
 use libra_types::validator_signer::ValidatorSigner;
 
+const HOST: &str = "http://localhost:8200";
+const ROOT_TOKEN: &str = "root_token";
+
 /// A test for verifying VaultStorage properly supports the SafetyRule backend.  This test
 /// depends on running Vault, which can be done by using the provided docker run script in
 /// `docker/vault/run.sh`
-#[ignore]
 #[test]
 fn test() {
+    let storage = Storage::from(VaultStorage::new(
+        HOST.to_string(),
+        ROOT_TOKEN.to_string(),
+        None,
+        None,
+    ));
+    if let Err(e) = storage.available() {
+        println!("Vault not available: {}", e.to_string());
```

Comment:
Do you know if this will actually show up? I was under the impression that tests only dump output on a failure, which won't be the case here?

Reply:
Yeah, I whined about the flaky test last night, I was told that it was actually disabled in master :D

println => nocapture -- sadly... I couldn't find anything in the spec that would make it obvious that the test did NOT run. I think we could explore an environmental variable that would return an error if vault didn't run.

@rexhoffman, what do you think?

LIBRA_REQUIRE_VAULT_TESTS and if set fail if available is err?

Reply:
IIRC This is what the language folks have been doing with their move prover tests. They have an environment variable that they set to "run" the tests. Same thing should be able to be applied here and if that envvar is set (which we would do in CI) and vault can't run, just panic to explicitly fail the test.

We probably want to think about the longer-term viability of this and if this is a maintainable solution but this would work in the short term.